src/Symfony/Component/Config/FileLocator.php

```diff
@@ -58,19 +62,28 @@ public function locate($name, $currentPath = null, $first = true)
         $paths = array_unique($paths);
         $filepaths = array();
 
-        foreach ($paths as $path) {
-            if (@file_exists($file = $path.DIRECTORY_SEPARATOR.$name)) {
-                if (true === $first) {
-                    return $file;
-                }
+        $load = function ($file) use (&$filepaths) {
+            if (@file_exists($file)) {
                 $filepaths[] = $file;
             }
+        };
+
+        foreach ($paths as $path) {
+            if (array() !== $files = glob($path.DIRECTORY_SEPARATOR.$name)) {
+                array_walk($files, $load);
+            } else {
+                $load($path.DIRECTORY_SEPARATOR.$name);
```

Comment:
I agree that this doesn't do anything.
Originally I wanted to preserve any existing functionality, but the `glob` should cover all current use cases.